### PR TITLE
Apply v1.3 API updates + remove JSON fields

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** 
Issues from source package.
https://github.com/fivetran/dbt_tiktok_ads_source/issues/8
https://github.com/fivetran/dbt_tiktok_ads_source/issues/7

**This PR will result in the following new package version:** 0.4.0

Although no fields were renamed due to the API upgrade, JSON array fields were removed to avoid BigQuery issues, so this is a breaking change.

**Please detail what change(s) this PR introduces and any additional information that should be known during the review of this PR:**
- In the [July 2023 connector update for TikTok Ads](https://fivetran.com/docs/applications/tiktok-ads/changelog), we upgraded the connector to the v1.3 API. As a result the dependent fields and field names from the downstream staging models have changed.
- None of the renamed fields were brought into the `dbt_tiktok_ads` package other than `age ` to `age_groups`, but it was subsequently removed due to an additional issue. We did need to update seed files with the renamed field names.
- Removed JSON array fields (`action_categories`, `age_groups`, `interest_category` and `language`) from our models that were causing issues with BigQuery, which currently does not support JSON arrays in group by functions.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt compile
- [x] dbt run –full-refresh
- [x] dbt run
- [x] dbt test
- [x] dbt run –vars (if applicable)

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked and tagged
- [x] You are assigned to the corresponding issue and this PR
- [x] BuildKite integration tests are passing

### Detailed Validation
Please acknowledge that the following validation checks have been performed prior to marking this PR as "ready for review":
- [x] You have validated these changes and assure this PR will address the respective Issue/Feature.
- [x] You are reasonably confident these changes will not impact any other components of this package or any dependent packages.
- [x] You have provided details below around the validation steps performed to gain confidence in these changes.

Fields mentioned above have been removed from the models in seed and development packages. Run off this branch to confirm. 

### Standard Updates
Please acknowledge that your PR contains the following standard updates:
- Package versioning has been appropriately indexed in the following locations:
    - [x] indexed within dbt_project.yml
    - [x] indexed within integration_tests/dbt_project.yml
- [x] CHANGELOG has individual entries for each respective change in this PR
    
- [x] README updates have been applied (if applicable)
    
- [NA] DECISIONLOG updates have been updated (if applicable)
- [x] Appropriate yml documentation has been added (if applicable)

### dbt Docs
Please acknowledge that after the above were all completed the below were applied to your branch:
- [x] docs were regenerated (unless this PR does not include any code or yml updates)

### If you had to summarize this PR in an emoji, which would it be?
 
🕙

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,9 +1,15 @@
-# dbt_tiktok_ads v0.UPDATE.UPDATE
+# dbt_tiktok_ads v0.4.0
+[PR #11](https://github.com/fivetran/dbt_tiktok_ads/pull/11) includes the following changes:
 
- ## Under the Hood:
+## 🚨 Breaking Changes 🚨:
+- In the [July 2023 connector update for TikTok Ads](https://fivetran.com/docs/applications/tiktok-ads/changelog), we upgraded the connector to the v1.3 API. As a result the dependent fields and field names from the downstream staging models have changed.
+- None of the renamed fields were brought into the `dbt_tiktok_ads` package other than `age ` to `age_groups`, but it was subsequently removed due to an additional issue. We did need to update seed files with the renamed field names.
+- Removed JSON array fields (`action_categories`, `age_groups`, `interest_category` and `language`) from our models that were causing issues with BigQuery, which currently does not support JSON arrays in group by functions
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:7]: I would request we make this a bit more concise as to what is changing inform users what exactly was removed, and a short explainer as to why. For example, there is probably no reason to explain the renaming of age to age_groups if we are just removing it already. My suggestion would be something along the lines of the following:
```suggestion
- In the [July 2023 connector update for TikTok Ads](https://fivetran.com/docs/applications/tiktok-ads/changelog), the connector was updated to support the TikTok Ads v1.3 API. As a result breaking changes exist within the dependent [v0.4.0 dbt_tiktok_ads_source](https://github.com/fivetran/dbt_tiktok_ads_source/releases/tag/v0.4.0) release in addition to the following breaking changes within this package release:
| **model** | **removed_fields** |
| ----------| -------------------- |
| [tiktok_ads__ad_group_report](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.tiktok_ads__ad_group_report) | `action_categories`, `age`, `languages`, `interest_category` |
| [tiktok_ads__ad_report](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.tiktok_ads__ad_report) | `action_categories`, `age`, `languages`, `interest_category` |
| [tiktok_ads__url_report](https://fivetran.github.io/dbt_tiktok_ads/#!/model/model.tiktok_ads.tiktok_ads__url_report) | `action_categories`, `age`, `languages`, `interest_category` |

>**Note**: Some of the above fields were also removed due to complications with the BigQuery JSON datatype causing errors during compliation.
```

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-avinash thanks for working through this API update! Your changes look good, I just have a few requests to update the code highlighted in my review below.

Additionally, one more change that needs to be applied before approving:
- Be sure to update the README version of the package as this is a breaking change. I noticed you marked that check as `NA` in the PR submission form. However, since this is a breaking change, we will want to make that update to reference the new version range.

File: CHANGELOG.md (hunk 1 above)
Comment (fivetran-avinash) [reply CHANGELOG.md:7]: Agreed, thanks for this suggestion.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-joemarkiewicz Thanks for reminding me about the README. Changes are applied as well.

File: packages.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
 packages:
-- package: fivetran/tiktok_ads_source
-  version: [">=0.3.0", "<0.4.0"]
+# - package: fivetran/tiktok_ads_source
+#   version: [">=0.3.0", "<0.4.0"]
+
+  - git: https://github.com/fivetran/dbt_tiktok_ads_source.git
+    revision: feature/tiktok-ads-v1.3-api-update
+    warn-unpinned: false
```
Comment (fivetran-joemarkiewicz) [inline packages.yml:7]: Usual reminder to update this before merge.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-avinash thanks for making the updates. Approved!

**Pull request merged**